modules/core/src/main/scala/com/azavea/stac4s/StacCatalog.scala

```diff
@@ -1,33 +1,82 @@
 package com.azavea.stac4s
 
 import cats.Eq
+import cats.implicits._
 import io.circe._
+import io.circe.syntax._
 
 final case class StacCatalog(
     stacVersion: String,
     stacExtensions: List[String],
     id: String,
     title: Option[String],
     description: String,
-    links: List[StacLink]
+    links: List[StacLink],
+    extensionFields: JsonObject
 )
 
 object StacCatalog {
 
+  val catalogFields = Set(
+    "stac_version",
+    "stac_extensions",
+    "id",
+    "title",
+    "description",
+    "links"
+  )
+
   implicit val eqStacCatalog: Eq[StacCatalog] = Eq.fromUniversalEquals
 
-  implicit val encCatalog: Encoder[StacCatalog] =
-    Encoder.forProduct6("stac_version", "stac_extensions", "id", "title", "description", "links")(catalog =>
+  implicit val encCatalog: Encoder[StacCatalog] = new Encoder[StacCatalog] {
+
+    def apply(catalog: StacCatalog): Json = {
+      val baseEncoder: Encoder[StacCatalog] =
+        Encoder.forProduct6("stac_version", "stac_extensions", "id", "title", "description", "links")(catalog =>
+          (
+            catalog.stacVersion,
+            catalog.stacExtensions,
+            catalog.id,
+            catalog.title,
+            catalog.description,
+            catalog.links
+          )
+        )
+
+      baseEncoder(catalog).deepMerge(catalog.extensionFields.asJson)
```

Comment:
I had a similar thought while working on this. I opted against it because it takes some effort to change all the stuff, we have to use `ConfiguredJsonCodec` in a few places because of `type` / `_type`, and we only get to use the derived codecs for things without the `extension:property` naming convention anyway.

We'll probably all laugh about this in 6 months, but I don't think we'll need to make big changes to the base codecs too frequently